plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -417,8 +417,9 @@ runCompiler moduleName opts expr = do
 
     -- Pir -> (Simplified) Pir pass. We can then dump/store a more legible PIR program.
     spirT <- flip runReaderT pirCtx $ PIR.compileToReadable pirT
-    let spirP = PIR.Program () . void $ spirT
-    when (_posDumpPir opts) . liftIO $ dumpFlat spirP "simplified PIR program" (moduleName ++ ".pir-simplified.flat")
+    let spirPNoAnn = PIR.Program () . void $ spirT
+        spirP = PIR.Program mempty . fmap getSrcSpans $ spirT
```

Comment:
Good point. I think this is an argument for the original implementation, in which the compiler takes the annotation type as an argument and can generate Plutus Core with different annotations. If the annotation type is hardcoded, then it should probably include the `Provenance`.

Reply:
A problem is `Provenance` has no default value (it doesn't have a `Monoid` instance).

Reply:
Since

> this is all going to have to be redone somehow once we do anything serious with the versions

which I totally agree, I think we can fix this at that point.

Reply:
> I think this is an argument for the original implementation, in which the compiler takes the annotation type as an argument and can generate Plutus Core with different annotations.

Why? I think we should just always produce the maximally informative version. `Provenance SrcSpans` is strictly more informative than just `SrcSpans`, I think, and you can always throw away the `Provenance` stuff.

> A problem is Provenance has no default value (it doesn't have a Monoid instance).

Where do we need that? For `applyProgram` I guess? You could fix that to only need `Semigroup` by `<>` ing the annotations on the versions instead of using `mempty` I guess. That function is already all hacks, this doesn't make it much worse!

Reply:
> which I totally agree, I think we can fix this at that point.

(this comment got into the wrong thread, but I totally agree, just wanted to register my feelings about `applyProgram` :sweat_smile: )

Reply:
happy to leave the provenance stuff for later, unless you want to get it out of the way while you're touching this code